### ISSUE TITLE
FR: Bulk deletion of bookmarks

### ISSUE THREAD

**Issue by Jellby** [koreader/koreader#8038]
**Does your feature request involve difficulty completing a task? Please describe.**
I use bookmarks mostly to highlight possible typos or mistakes in a book that I later fix on the computer. When I'm fixing a book a go through the list of bookmarks, and once I've fixed or verified them, I delete them.

The problem is that after I've checked all 14-15 bookmarks in a page (a page in the bookmarks list, not in the book), I have to delete them one by one, and each of them involves a sequence of long tap, tap, tap, which gets annoying. I'd like something more convenient.

**Describe the solution you'd like**
Best (for me) would be to have an option (how to access it? I don't know) to delete all bookmarks currently appearing on the screen.

Second best: add a checkbox for each bookmark, so I can toggle them one by one, and then an option to delete all checked bookmarks (in current screen? In the whole list?).

**Describe alternatives you've considered**
An option to delete all bookmarks in a book would not be so good. Sometimes I have many corrections to do and work on them on several sessions.

**Comment by jlynton** [comment]
I have the same problem as Jellby, and would find one of the solutions that he proposes very useful indeed!

**Comment by hius07** [comment]
Can it be a 'Delete mode'?
If switched on, tapping the bookmark in the list would remove it without confirmation.

**Comment by Frenzie** [comment]
While I don't really understand the use case for deleting all bookmarks on the screen, tapping 15-20 times doesn't sound like it would help all that much. :-) The "second best" seems like it'd be the best, imho. (Because it's generic, not specific.)

**Comment by Jellby** [comment]
> tapping 15-20 times doesn't sound like it would help all that much.

It would be much better than the current method, that requires long tapping and taping in three different locations for each bookmark, a minimum of 66% time saving (only one third of the taps), probably quite a bit more (accounting for long taps and changing locations). A checkbox ("second best" solution) still requires tapping 15-20 times, but I agree it's more generic (it could be used for other things in a future, like bulk renaming, changing properties, exporting...).

> While I don't really understand the use case for deleting all bookmarks on the screen

In case that's a genuine question, and not just a way of saying "I don't need that feature", let me elaborate:

While I'm reading a book, I highlight things that need double checking, like obvious typos, words or expressions of dubious correctness, places with wrong or inconsistent format, etc. These are my only bookmarks, I don't bookmark (in general) passages for later reference or long-term storage.

Once I finish the book, now at a computer, I go through my list of bookmarks. The highlighted text, which appears as the bookmark "title" is usually enough for me to (a) locate it in the book's source files, and (b) see or remember what was "wrong" with it, so I don't need to see the bookmark in the book, just the list is enough. For each bookmark in the page (of bookmarks list), I fix it in the book's source if necessary (e.g., compare with printed version, scans, decide if it's a mistake). Once I've checked all the bookmarks in a page, I don't need them any more. I could just go to next/previous page, but if there are e.g. 20 pages of bookmarks (not that uncommon), I won't probably fix them all in a single session, and I'd like to continue next time where I left. The easiest way is to simply delete the bookmarks that I have already "fixed". And since I work on them page by page, the fastest would be an action to delete all bookmarks in the page.

So, in short, my workflow is:
* Open bookmarks list
* Go to last page (because they are sorted from newest to oldest)
* Start fixing from the bottom of the page to the top
* Once I reach the top one, _delete all the bookmarks in the page_ (currently, this has to be done one by one, and it gets annoying)
* Go to previous page (this happens automatically, since I've deleted all bookmarks in the last page)

**Comment by Frenzie** [comment]
> In case that's a genuine question, and not just a way of saying "I don't need that feature", let me elaborate:

It's neither, but thanks for the elaborate reply
 ^_^

> A checkbox ("second best" solution) still requires tapping 15-20 times, but I agree it's more generic

I'm not sure which checkboxes you've been using but anything that deals with more than half a dozen or so elements should cone with check/uncheck all. ;-)

**Comment by hius07** [comment]
(Not touching the menu widget) it might be something like:

New button <kbd>Remove more bookmarks</kbd>

<kbd>![1](https://user-images.githubusercontent.com/62179190/136539241-99f0f34d-ee2e-4fc6-8dd7-4ba6ce90fb30.png)</kbd>

Tap to select/unselect a bookmark

<kbd>![2](https://user-images.githubusercontent.com/62179190/136539328-e6b65025-366d-47f9-8498-de8cf013e1e5.png)</kbd>

Selected items are dimmed (as deleted books in the History)

<kbd>![3](https://user-images.githubusercontent.com/62179190/136539397-561bd8f2-ef0b-44ad-a6b1-394e46a579c1.png)</kbd>

Then long-press any item

<kbd>![4](https://user-images.githubusercontent.com/62179190/136539496-c1e7eab3-de35-41fa-91fb-288142bf8801.png)</kbd>

**Comment by hius07** [comment]
To discuss proposed changes:
-Multiple removing: select single / page / all
-Filtering by categories / Search for a string in highlights/notes

![17](https://user-images.githubusercontent.com/62179190/138272848-9c0e918a-a227-4e7e-a39d-e2d9c7f1c176.png)

![2](https://user-images.githubusercontent.com/62179190/138273217-d75f3ce5-4002-4f01-8eeb-6af233abaea5.png)

![3](https://user-images.githubusercontent.com/62179190/138272913-8df677bd-ccd4-4a61-a998-a0014f79f124.png)

![12](https://user-images.githubusercontent.com/62179190/138272996-68c2a04d-5ab1-4d8e-bf6b-617e369e3150.png)
